web-bundle/src/main/java/com/graphhopper/resources/InfoResource.java

```diff
@@ -104,16 +103,19 @@ public Info getInfo() {
 
             List<Object> possibleValueList = new ArrayList<>();
             try {
-                Class<? extends Enum> enumClass = evFactory.findValues(encodedValue);
-                for (Object o : enumClass.getEnumConstants()) {
+                EnumEncodedValue eev = storage.getEncodingManager().getEncodedValue(encodedValue, EnumEncodedValue.class);
+                for (Object o : eev.getValues()) {
                     possibleValueList.add(o.toString());
                 }
-            } catch (IllegalArgumentException ex) {
-                // we expect BooleanEncodedValue here
-                possibleValueList.add(true);
-                possibleValueList.add(false);
+            } catch (ClassCastException e) {
```

Comment:
This is really ugly. Why can I not obtain an encoded value from EncodingManager via the name **without** having it cast to something, when I do not even know the type. Hm, or should I just use `getEncodedValue(encodedValue, Object.class)`, but that still seems confusing to me.

Reply:
Btw there should also be a method to be able to iterate over all encoded values?

Reply:
> Hm, or should I just use getEncodedValue(encodedValue, Object.class), but that still seems confusing to me

The problem is that you really have to know the type (boolean, int, decimal or enum). With just EncodedValue you cannot do much. I.e. `edge.get` requires a specific EncodedValue to force a fast access without the need for casting. But maybe there is a better and still fast solution for this.

Reply:
> Or can we just add a setter for the encodedValueMap?

You mean in the EncodingManager? There should be only the possibility at creation time to add EncodedValues.

> Btw there should also be a method to be able to iterate over all encoded values?

Yeah, probably. Something similar to fetchEdgeEncoders

Reply:
`if(ev = getEncodedValue(encodedValue, EncodedValue.class) instanceof EnumEncodedValue)` should work

Reply:
> Or can we just add a setter for the encodedValueMap?

Sorry I meant **getter*

Reply:
> em.getEncodedValue(encodedValue, EncodedValue.class)

Ok yes this works (no class cast exception), but why not just `em.getEncodedValue(encodedValue)` and then I can do the casting myself, do an instanceof check etc.

Reply:
> Sorry I meant getter

Hmmh, but at some point you will always need a cast to use it. (Ok, not in this specific InfoResource use case, but here we could workaround via the snippet above?)

Reply:
Yes I did. But can we not get also rid of the list of hard-coded encoded values above? How do we know which ones shall be available for auto complete and which ones should not? Can we just take **all** enum- and boolean-encoded values from encoding manager (for this the getter or some iterator would be needed).

Reply:
Sorry, github does not really update this is so ugly ... have not seen your latest comment:

>  but why not just em.getEncodedValue(encodedValue) and then I can do the casting myself, do an instanceof check etc

The interface currently makes it very clear that this is not a "generic" thing. When we introduce the getter without the class (We can do that via just removing the class parameter in the "generic" get method), then it might happen that it is unclear what to do with an EncodedValue or might be even seen as a bug that the generic `edge.get(EncodedValue)` is not possible.

Reply:
> When we introduce the getter without the class

Ok, `em.getEncodedValue(encdodValue, EncodedValue.class)` is good enough for me :+1: (but I would still like to get rid of the hard-coded list of encoded values available for auto-complete if possible)